src/py21cmfast/src/Stochasticity.c

```diff
@@ -907,6 +900,15 @@ int sample_halo_grids(gsl_rng **rng_arr, double redshift, float *dens_field,
         istart_threads[threadnum] = istart;
         nhalo_threads[threadnum] = count;
     }
+    if (OUT_OF_BUFFER) {
+        LOG_ERROR("More than %llu halos (expected %.1e) with buffer size factor %.1f",
+                  arraysize_local, arraysize_local / config_settings.HALO_CATALOG_MEM_FACTOR,
+                  config_settings.HALO_CATALOG_MEM_FACTOR);
+        LOG_ERROR(
+            "If you expected to have an above average halo number try raising "
+            "config['HALO_CATALOG_MEM_FACTOR']");
+        Throw(ValueError);
+    }
```

Comment:
In order to avoid some confusion (and help #550), it might be a good idea to output counts in each thread in the error message, or at least refer to which thread had the overflow